### PR TITLE
feat: mocha -> vitest filemod

### PR DESCRIPTION
**Pull request by r4zendev**

### CODE REVIEW

File: codemods/mocha/vitest/config-files/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+		'**/tsconfig.json',
+		'**/{,.}{mocharc,mocha.config}{,.js,.json,.cjs,.mjs,.yaml,.yml}',
+	],
+	excludePatterns: ['**/node_modules/**'],
+	handleFile: async (_, path, options) => {
+		if (path.endsWith('tsconfig.json') || path.endsWith('package.json')) {
+			return [{ kind: 'upsertFile', path, options }];
+		}
+
+		// mocharc
```
Comment (grzpab) [inline codemods/mocha/vitest/config-files/src/index.ts:15]: this should be checked explicitly

File: codemods/mocha/vitest/config-files/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+		return [{ kind: 'deleteFile', path }];
+	},
+	handleData: async (_, path, data) => {
+		if (path.endsWith('package.json')) {
+			const json = JSON.parse(data);
```
Comment (grzpab) [inline codemods/mocha/vitest/config-files/src/index.ts:20]: I'd do it with a try / catch

File: codemods/mocha/vitest/config-files/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+			// Remove possible "mocha" key and its value
+			if (json.mocha) {
```
Comment (grzpab) [inline codemods/mocha/vitest/config-files/src/index.ts:22]: can we use valibot for that?

File: codemods/mocha/vitest/config-files/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+					...json.scripts,
+					test: 'vitest run',
+					coverage: 'vitest run --coverage',
+				};
+			}
+
+			return {
+				kind: 'upsertData',
+				path,
+				data: JSON.stringify(json, null, 2),
```
Comment (grzpab) [inline codemods/mocha/vitest/config-files/src/index.ts:66]: why the remaining 2 arguments? for pretty print?

Comment (r4zendev) [reply codemods/mocha/vitest/config-files/src/index.ts:66]: yes

**Comment by r4zendev** [comment]
image

With valibot bundle size remained almost the same, so we should be good. @grzpab

**Review by grzpab — APPROVED** [review]

**Pull request merged**